src/iam.ts

```diff
@@ -272,6 +277,27 @@ export class IAM extends IAMBase {
     throw new Error(ERROR_MESSAGES.DID_DOCUMENT_NOT_INITIALIZED);
   }
 
+  async updateAssetDidDocument(options: {
+    did: string,
+    value: IAttributePayload
+  }) {
+    const { did, value } = options;
+    if (did && this._didSigner) {
+      const defaultValue: IAttributePayload = { publicKey: `0x${new Keys().publicKey}` };
+      const updateData: IUpdateData = {
+        algo: Algorithms.Secp256k1,
+        type: PubKeyType.SignatureAuthentication2018,
+        encoding: Encoding.HEX,
+        value: {...defaultValue, ...value },
+      };
+      const operator = new ProxyOperator(this._didSigner, this._registrySetting, addressOf(did));
+      const update = await operator.update(did, DIDAttribute.PublicKey, updateData);
+      return Boolean(update);
```

Comment:
Could you please explain what is purpose of returning boolean?

Reply:
I was looking on implementation `updateDidDocument` method and creating this one similarly. You think it would be better to return `Promise<utils.BigNumber>` which is returned by update method?